tests/ui/rfcs/rfc-0000-never_patterns/ICE-133117-duplicate-never-arm.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+
+fn main() {
+    match () {
+        (!|
+        //~^ ERROR: mismatched types
+        !) if true => {} //~ ERROR a never pattern is always unreachable
+        //~^ ERROR: mismatched types
+        (!|!) if true => {} //~ ERROR a never pattern is always unreachable
+    }
+}
```

Comment:
Same here, could you make the test type-correct:
```rust
#![feature(never_type)]
#![feature(never_patterns)]
#![allow(incomplete_features)]

fn foo(x: !) {
    match x {
        (!|!) if true => {}
        (!|!) if true => {}
    }
}
```

Reply:
Fixed in https://github.com/rust-lang/rust/pull/135409/commits/e7db0925d1441990ceb8236b5d4015f9a2ba333c.